scripts/hudson/narayana.sh

```diff
@@ -353,53 +353,37 @@ function build_narayana {
 }
 
 function clone_as {
-  echo "Cloning AS sources from https://github.com/jbosstm/jboss-as.git"
+  repo="https://github.com/wildfly/wildfly.git" # was https://github.com/jbosstm/jboss-as.git
+  AS_BRANCH="26.x"
+  echo "Cloning AS sources from $repo"
 
   cd ${WORKSPACE}
   if [ -d jboss-as ]; then
     echo "Updating existing checkout of WildFly"
     cd jboss-as
 
-    git remote | grep upstream
-    if [ $? -ne 0 ]; then
-      git remote add upstream https://github.com/wildfly/wildfly.git
-    fi
     #Abort any partially complete rebase
     git rebase --abort
-    git checkout 5_BRANCH
-    [ $? -eq 0 ] || fatal "git checkout 5_BRANCH failed"
+    git checkout $AS_BRANCH
+    [ $? -eq 0 ] || fatal "git checkout $AS_BRANCH failed"
     git fetch
-    [ $? -eq 0 ] || fatal "git fetch https://github.com/jbosstm/jboss-as.git failed"
-    git reset --hard jbosstm/5_BRANCH
-    [ $? -eq 0 ] || fatal "git reset 5_BRANCH failed"
+    [ $? -eq 0 ] || fatal "git fetch $repo failed"
     git clean -f -d -x
     [ $? -eq 0 ] || fatal "git clean failed"
     git rebase --abort
     rm -rf .git/rebase-apply
   else
     echo "First time checkout of WildFly"
-    git clone https://github.com/jbosstm/jboss-as.git -o jbosstm
-    [ $? -eq 0 ] || fatal "git clone https://github.com/jbosstm/jboss-as.git failed"
+    git clone https://github.com/wildfly/wildfly.git jboss-as
+    [ $? -eq 0 ] || fatal "git clone $repo failed"
 
     cd jboss-as
-
-    git remote add upstream https://github.com/wildfly/wildfly.git
+    git checkout $AS_BRANCH
   fi
+  git remote add upstream https://github.com/wildfly/wildfly.git
```

Comment:
This is an interesting change, the implication of no "-d jboss-as" is that the upstream remote would not exist and so we add the upstream remote if the directory was not present but it's not guaranteed that the directory would exist and the remote would not. If you want to move from the previous assumption I think you are going to need to set this adding of the remote into something like:
```git remote -v | grep upstream | grep https://github.com/wildfly/wildfly.git
  if [ $? = 1 ]; then
    git remote add upstream https://github.com/wildfly/wildfly.git
  fi````

Reply:
sorry I think I have misread this block

Reply:
I think is still be generally relevant but I had misread a bit and now I realised 386 was in the block where the directory did not exist )though as commented higher up around 363 there should be handling for the case where there may or may not be an "upstream" remote already

Reply:
We need an upstream so that we can rebase against 26.x hence `git remote add upstream https://github.com/wildfly/wildfly.git`.

Reply:
My point is that repeatedly running this script (say locally) would try to do the `git remote add upstream https://github.com/wildfly/wildfly.git` again and that should be protected. If you just run it without trying to see if it might be there already:
```
$ git remote add upstream https://github.com/wildfly/wildfly.git
error: remote upstream already exists.
```